### PR TITLE
Chat completion: messages, stop, temperature, top_p, model, max_completion_tokens

### LINKED ISSUE #1: messages
As a API user I want to provide `/v1/chat/completions` endpoint with an initial list of messages.
Format described there:
https://platform.openai.com/docs/api-reference/chat
https://github.com/openai/openai-openapi

Document required model configuration to support `messages` (e.g. chat template).
Make sure underlying completion model receives correctly formatted text (according to model configuration).
Assert model is of "instruct" type.(???)

Update OpenAPI file to correctly represent the parameter.

Invariants to test:
- schema
- different roles ("user", "system", "assistant")
- conversion from "chat completion" into "text completion" and backwards.

### PR DESCRIPTION
**Pull request by vidas**
Implement several "low hanging fruit" to align `/v1/chat/completions` arguments with OpenAI API.

Most were already available, just needed verification, minor tweaking and testing.

Breaking change - `model` is now mandatory.

Closes #1, #2, #3.

### CODE REVIEW

File: llama_cpp/server/app.py [hunk 1]
```diff
@@ -504,14 +495,9 @@ async def create_chat_completion(
...
-        _min_tokens_logits_processor = llama_cpp.LogitsProcessorList(
-            [llama_cpp.MinTokensLogitsProcessor(body.min_tokens, llama.token_eos())]
-        )
-        if "logits_processor" not in kwargs:
-            kwargs["logits_processor"] = _min_tokens_logits_processor
-        else:
-            kwargs["logits_processor"].extend(_min_tokens_logits_processor)
+    # Override max_tokens with max_completion_tokens.
+    if body.max_completion_tokens is not None:
+        kwargs["max_tokens"] = body.max_completion_tokens
```
Comment (akxcv) [inline llama_cpp/server/app.py:500]: It's best to use `max_completion_tokens` in `kwargs` and throughout the whole codebase to keep the terminology consistent, if it's not too complicated to change right now.

**Review by akxcv — COMMENTED** [review]
It would be perfect if you could add a couple of tests covering the use of these parameters.

**Comment by vidas** [comment]
@akxcv I had some test, but decided to intentionally omit them (have some in another branch) to later add once #5 is done, so that everything is consistent.

**Comment by akxcv** [comment]
Sounds good! Could you create an issue for covering these parameters with tests, so we won't forget?

File: llama_cpp/server/app.py (hunk 1 above)
Comment (vidas) [reply llama_cpp/server/app.py:500]: `max_tokens` is used pervasively throughout the code. While I could globally search/replace it, I want to limit our edits to `llama_cpp/server/*` if possible. I still believe, that we will be able to include python bindings as a dependency (i.e. use llama-cpp-python as a library directly from upstream) and only have "Server" part custom. I don't think we are ready to maintain bindings to llama.cpp ourselves.

Comment (akxcv) [reply llama_cpp/server/app.py:500]: In that case, yes, let's keep it as is!

**Review by akxcv — APPROVED** [review]

**Pull request merged**